reframe/core/config.py

```diff
@@ -217,3 +220,151 @@ def create_env(system, partition, name):
                 system.add_partition(part)
 
             self._systems[sys_name] = system
+
+
+def ppretty(value, htchar=' ', lfchar='\n', indent=4, current_depth=0):
+    nlch = lfchar + htchar * indent * (current_depth + 1)
+    if type(value) is dict:
+        if value == {}:
+            return '{}'
+        items = [
+            nlch + repr(key) + ': ' +
+            ppretty(value[key], htchar, lfchar, indent, current_depth + 1)
+            for key in value
+        ]
+        return '{%s}' % (','.join(items) + lfchar +
+                         htchar * indent * current_depth)
+    elif type(value) is list:
```

Comment:
```suggestion
    elif isinstance(value, collections.abc.Sequence):
```

Reply:
The problem is that strings are also recognized as a `collections.abc.Sequence`, which leads to an infinite loop. So I think I should leave it as
```suggestion
    elif isinstance(value, list):
```

Reply:
Ok, so keep your original checks everywhere (dict, list, tuple) for consistency. You should also have a case for sets.

Reply:
What should I print in the case of an empty set? `set()`?

Reply:
Yes.